webapp/channels/src/components/post_view/post_reaction/post_reaction.test.tsx

```diff
@@ -16,24 +18,62 @@ describe('components/post_view/PostReaction', () => {
         getDotMenuRef: jest.fn(),
         showIcon: false,
         showEmojiPicker: false,
-        toggleEmojiPicker: jest.fn(),
+        setShowEmojiPicker: jest.fn(),
         actions: {
             toggleReaction: jest.fn(),
         },
     };
 
-    test('should match snapshot', () => {
-        const wrapper = shallowWithIntl(<PostReaction {...baseProps}/>);
-        expect(wrapper).toMatchSnapshot();
+    const userId = 'userId';
+    const initialState = {
+        entities: {
+            roles: {
+                roles: {
+                    system_user: TestHelper.getRoleMock({permissions: [Permissions.ADD_REACTION]}),
```

Comment:
I really want to make a better way for setting up these tests with permissions since it was annoying trying to figure out that the missing permissions were why this component didn't render

Reply:
The goal would be to make something reusable to avoid having to type the Redux store out, with or without AI 😛 . For a while now, I've wanted to either autopopulate more of the Redux state (like provide a default current user/team/channel/config/roles) or to extend `TestHelper` with something that can build a Redux state more like this:
```js
const builder = new StateBuilder();

const testUser = builder.createUser({username: 'test_user'}, {asCurrentUser: true});
const testChannel = builder.createChannel({type: 'P', display_name});

renderWithContext(<MyComponent channelId={testChannel.id}/>, builder.getState());
```